### PR TITLE
SDK for macOS (include catalyst)

### PR DESCRIPTION
**Pull request by haoliu-amp**
- Added osx option to podspec
- Added osx check for UIKit for source files

### CODE REVIEW

**Comment by haoliu-amp** [comment]
@rob-keepsafe This is the change I did for macOS support. Sorry not able to directly merge yours because there're way too many conflicts since it's a change a few years ago (definitely our fault). 

If you want, you can take a look to see if I miss anything. Thanks!

**Comment by rob-keepsafe** [comment]
@haoliu-amp Looks good at a high level but I haven't compiled/verified anything since we aren't planning on supporting macOS anymore

It might be worth reaching out to the people on https://github.com/amplitude/Amplitude-iOS/pull/155 or https://github.com/amplitude/Amplitude-iOS/issues/138

File: Amplitude-iOS.podspec [hunk 1]
```diff
@@ -6,8 +6,9 @@ Pod::Spec.new do |s|
   s.license                = { :type => "MIT" }
   s.author                 = { "Amplitude" => "dev@amplitude.com" }
   s.source                 = { :git => "https://github.com/amplitude/Amplitude-iOS.git", :tag => "v4.10.0" }
-  s.ios.deployment_target  = '8.0'
+  s.ios.deployment_target  = '13.2'
   s.tvos.deployment_target = '9.0'
+  s.osx.deployment_target  = '10.15'
```
Comment (nikitakirichek) [inline Amplitude-iOS.podspec]: Does this mean that the minimum OS X deployment target is 10.15? Can we make it to be 10.10?

**Comment by nikitakirichek** [comment]
It will be also nice to have Amplitude library as a target, not only framework.

**Comment by nikitakirichek** [comment]
I can make a PR to **mac_app_support** branch with 10.10 support and adding a library, if you find it suitable.

**Comment by haoliu-amp** [comment]
@nikitakirichek Sure, would like to see it too!

**Comment by haoliu-amp** [comment]
> It will be also nice to have Amplitude library as a target, not only framework.

I'm actually considering even removing our library target. What's the reason that you wanna have library versus framework?

**Comment by nikitakirichek** [comment]
@haoliu-amp There is no reason to have library in my project, so no need to have it in amplitude too.

**Comment by haoliu-amp** [comment]
Hey folks! Now v5.0.0 supports macOS instrumentation. Hope you can try it out and let us know if there're any issues!

**Pull request merged**